src/plugins/intel_gpu/src/graph/impls/ocl/non_max_suppression.cpp

```diff
@@ -127,6 +129,12 @@ struct non_max_suppression_impl : typed_primitive_impl_ocl<non_max_suppression>
             params.has_third_output = true;
         }
 
+        if (arg.use_multiple_outputs()) {
```

Comment:
Please get rid of usage of "arg"

Reply:
In case of `non_max_suppression`, there are many parts where APIs of `typed_program_node` are used during primitive_impl creation. IMO, it should be handled as a separate task to remove program_node dependency in primitive_impl creation parts.

Reply:
yes, but it will be better to get rid of newly added ones.